mitosheet/src/components/taskpanes/SnowflakeImport/SnowflakeImportTaskpane.tsx

```diff
@@ -0,0 +1,478 @@
+                            {availableSnowflakeOptionsAndDefaults.type === 'error' && availableSnowflakeOptionsAndDefaults.error_message}
+                        </div>
+                    }
+                </CollapsibleSection>
+                <Spacer px={20}/>
+                <CollapsibleSection title="Connection Location" open={availableSnowflakeOptionsAndDefaults?.type === 'success'}>
+                    <Row justify="space-between">
+                        <Col>
+                            Warehouse
+                        </Col>
+                        <Col>
+                            <Select
+                                width="medium"
+                                value={params.connection.warehouse || 'None available'}
+                                onChange={(newWarehouse) => {
+                                    setParams((prevParams) => {
+                                        return {
+                                            ...prevParams, 
+                                            connection: {
+                                                ...prevParams.connection, 
+                                                warehouse: newWarehouse
+                                            }
+                                        }
+                                    })
+                                }}
+                            >
+                                {availableSnowflakeOptionsAndDefaults?.type === 'success' ? availableSnowflakeOptionsAndDefaults.config_options.warehouses.map((warehouse) => {
+                                    return (
+                                        <DropdownItem key={warehouse} id={warehouse} title={warehouse}/>
+                                    )
+                                }) : []}
+                            </Select>
+                        </Col>
+                    </Row>
+                    <Row justify="space-between">
+                        <Col>
+                            Database
+                        </Col>
+                        <Col>
+                            <Select
+                                width="medium"
+                                value={params.connection.database || 'None available'}
+                                onChange={(newDatabase) => {
+                                    if (newDatabase === params['connection']['database']) {
+                                        return 
+                                    }
+                                    
+                                    const paramsCopy: SnowflakeImportParams = {...params}
+                                    const newParams = {
+                                        ...paramsCopy, 
+                                        'connection': {
+                                            ...paramsCopy.connection,
```

Comment:
There is a huge chunk of duplicated code between the database, schema, and table updaters. Let's write a little helper that at least takes a db, option takes a schema, table, etc -- and then sets in the obvious way!

There's generally a lot of code in this taskpane that I think can be refactored into well-named helper functions - and I think we should do this so it's not a 400 line monsta